src/config.ts

```diff
@@ -313,3 +305,54 @@ export const SOFTWARE_VERSION = (() => {
     // into <pre> or <code> where it will create an unnecessary newline.
     return /^(.*)$/m.exec(versionFile)?.at(0) ?? defaultText;
 })();
+
+
+// Command line related functions
+
+/**
+ * Grabs an argument from the command line and checks if it exists.
+ * @param arg Argument name
+ * @returns True if the argument is present, otherwise false.
+ */
+function getCommandlineArgumentPresent(arg: string): boolean {
+    return process.argv.includes(arg);
+}
+
+/**
+ * Grabs an argument's value from program arguments if it exists, otherwise returns undefined.
+ * @param arg Argument name
+ * @param throwOnInvalid If true or undefined, throws an error if the argument is present but has no value. If false, returns undefined.
+ * @returns value passed to the argument or undefined.
+ * @throws Error if the argument is present but has no value.
+ */
+function getCommandlineStringArgument(arg: string, throwOnInvalid?: boolean | undefined): string | undefined {
+    // we don't want to throw if the argument is not present
+    if(!getCommandlineArgumentPresent(arg)) return undefined;
+
+    const index = process.argv.indexOf(arg);
```

Comment:
The reason why previously this was previously called `configOptionIndex` is so that it is explicit to the reader what the index is for. Please don't erase verbose names like this, they are there for a reason.